sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -268,6 +268,12 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
   object CartesianProduct extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+      case logical.Join(CanBroadcast(left), right, joinType, condition) =>
+        execution.joins.BroadcastNestedLoopJoin(
+          planLater(left), planLater(right), joins.BuildLeft, joinType, condition) :: Nil
```

Comment:
ah, sorry. I missed that line. For a CROSS JOIN, we will use `INNER` as the join type and the condition should be `None`. Can explicitly match these? I mean to let others easy to understand the code, `case logical.Join(CanBroadcast(left), right, JoinType.Inner, None)` will be better, right?